sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -1573,9 +1573,9 @@ def get_blob_client(
                 "Please use 'BlobProperties.name' or any other str input type instead.",
                 DeprecationWarning
             )
-        try:
-            blob_name = blob.get('name')  # type: ignore
-        except AttributeError:
+        if hasattr(blob, 'name'):
```

Comment:
This check isn't quite right and is the reason you needed the type ignore below. We are checking for an attr `name` but then calling `get`.

I think this was here to handle `BlobProperties`? Can probably use `isinstance(BlobProperties)` instead. Even combine with first if probably.